rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/domain/model/LongRunningAction.java

```diff
@@ -725,24 +725,33 @@ protected LRAStatus toLRAStatus(int atomicActionStatus) {
     }
 
     public LRAParticipantRecord enlistParticipant(URI coordinatorUrl, String participantUrl, String recoveryUrlBase,
-                                                  long timeLimit, String compensatorData) throws UnsupportedEncodingException {
-        LRAParticipantRecord participant = findLRAParticipant(participantUrl, false);
-
-        if (participant != null) {
-            participant.setCompensatorData(compensatorData);
-            return participant; // must have already been enlisted
+            long timeLimit, String compensatorData) throws UnsupportedEncodingException {
+        ReentrantLock lock = tryLockTransaction();
+        if (lock == null) {
+            LRALogger.i18nLogger.error_enlistment();
+            return null;
         }
-
-        participant = doEnlistParticipant(coordinatorUrl, participantUrl, recoveryUrlBase,
-                timeLimit, compensatorData);
-
-        if (participant != null) {
-            // need to remember that there is a new participant
-            deactivate(); // if it fails the superclass will have logged a warning
-            savedIntentionList = true; // need this clean up if the LRA times out
+        else {
```

Comment:
I don't think any of the following 17 lines of code have changed apart from the indentation has it?

Reply:
I can count 14 lines that haven't changed (from line 736 to 749) apart from the indentation. The finally block has been added in this PR.